README.md

```diff
@@ -169,6 +169,50 @@ summary.get(labels: { service: 'database' })
 # => { 0.5 => 0.1233122, 0.9 => 3.4323, 0.99 => 5.3428231 }
 ```
 
+## Labels
+
+All metrics can have labels, allowing grouping of related time series.
+
+Labels are an extremely powerful feature, but one that must be used with care.
+Refer to the best practices on [naming](https://prometheus.io/docs/practices/naming/) and 
+[labels](https://prometheus.io/docs/practices/instrumentation/#use-labels).
+
+Most importantly, avoid labels that can have a large number of possible values (high 
+cardinality). For example, an HTTP Status Code is a good label. A User ID is **not**.
+
+Labels are specified optionally when updating metrics, as a hash of `label_name => value`.
+Refer to [the Prometheus documentation](https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels) 
+as to what's a valid `label_name`.
+
+In order for a metric to accept labels, their names must be specified when first initializing 
+the metric. Then, when the metric is updated, all the specified labels must be present.
+
+Example:
+
+```ruby
+https_requests_total = Counter.new(:http_requests_total, docstring: '...', labels: [:service, :status_code])
+
+# increment the counter for a given label set
+https_requests_total.increment(labels: { service: "my_service", status_code: response.status_code })
+```
+
+### Pre-set Label Values
+
+You can also "pre-set" some of these label values, if they'll always be the same, so you don't
+need to specify them every time:
+
+```ruby
+https_requests_total = Counter.new(:http_requests_total, 
+                                   docstring: '...', 
+                                   labels: [:service, :status_code],
+                                   preset_labels: { service: "my_service" })
+
+# increment the counter for a given label set
+https_requests_total.increment(labels: { status_code: response.status_code })
+```
+
+
+
```

Comment:
It would *certainly* be a major version bump...
If we were above `1.0` :)
Semver says that below 1.0, anything goes (or at least, that's my understanding, I may be wrong)

But yes, this is definitely not backwards compatible. I've tried to keep it as compatible as possible, but there's no way to keep backwards compatibility *and* follow the current best practices.
That said, I've tried to minimize pain for existing consumers, while also making it better for future ones. 
Best practices suggest the method be called `inc`, for example. I chose to leave it as `increment`, because it felt like a needless contract break. But i also change the signature because the new one is better for the case we're supposed to cater the most for, which is no labels.

Reply:
Taking note of the idea of an upgrading section, and marking this thread as resolved.